runtime/near-vm-logic/src/logic.rs

```diff
@@ -1263,30 +1263,29 @@ impl<'a> VMLogic<'a> {
             }
         };
 
-        let mut pubkeys: Vec<blst::min_pk::PublicKey> = vec![];
-        for i in 0..pubkeys_cnt {
+        let mut pubkeys: Vec<blst::min_pk::PublicKey> =
+            Vec::with_capacity(num_pubkeys.try_into().unwrap());
+
+        for i in 0..num_pubkeys {
             pubkeys.push(
                 match blst::min_pk::PublicKey::key_validate(
                     &pubkeys_raw[((i * PUBKEY_LEN) as usize)..(((i + 1) * PUBKEY_LEN) as usize)],
                 ) {
                     Ok(pubkey) => pubkey,
                     Err(err) => {
-                        return Err(VMLogicError::from(Bls1238VerifyError(Bls12381Error::from(err))))
+                        return Err(VMLogicError::from(Bls1238VerifyError(Bls12381Error::from(
+                            err,
+                        ))))
                     }
                 },
             );
         }
 
-        let mut pubkeys_refs: Vec<&blst::min_pk::PublicKey> = vec![];
-        for i in 0..pubkeys.len() {
-            pubkeys_refs.push(&pubkeys[i]);
-        }
-
         let res = aggregate_sig.fast_aggregate_verify(
             true,
             &message,
             b"BLS_SIG_BLS12381G2_XMD:SHA-256_SSWU_RO_POP_",
-            pubkeys_refs.as_slice(),
+            &pubkeys.iter().collect::<Vec<_>>(),
```

Comment:
Having to allocate an additional `Vec` here seems unnecessary. Could the type of `fast_aggregate_verify` be changed so that it could accept a slice of keys or key pointers? Maybe something like `&[T] where T: AsRef<PublicKey>`?

Reply:
Unfortunately, this method is part of blst [lib](https://github.com/supranational/blst/blob/master/bindings/rust/src/lib.rs#L1006) , I will try to create a pull request to improve it, but not sure that this will be quickly accepted.